src/main/groovy/val/Idator.groovy

```diff
@@ -1,18 +1,75 @@
 package val
 
 /**
- * Serves as the entry point to validating an object graph. This is effectively a factory for the
- * initial DefinitionCheck which can begin validation.
+ * Represents a Check that is created through a "Definition": a closure using DSL style syntax to
+ * define the validations that should be performed on an input graph.
+ *
+ * The initial instance will normally be created by the Idator and then further instances will be created
+ * internally to form a graph of Checks which conceptually represents the structure of the input that will be validated
+ * and onto which the input graph will map.
+ *
+ * Idators will normally create a structure aligned with the expected input graph where each
+ * Idator will similarly be composed of those beneath it with a single DefinitionCheck attached to
+ * the root of the input graph and containing all other DefinitionChecks (normally resulting in an arborescence)
  */
-class Idator {
+class Idator implements Check {
 
     /**
-     * Start defining Checks at the root of the input graph
-     * @param definition A Closure which will be evaluated against the root DefinitionCheck of the input graph
-     * @return The Check created by definition
+     * A shared instance of Checkers which is used as a delegate for Definitions to allow Check construction
+     * (and used internally)
      */
-    Check define(Closure definition) {
-        new DefinitionCheck(new val.Checkers(), [:], 'root', definition)
+    @Delegate final Checkers checkers
+
+    /**
+     * A Map of stashed values shared among the graph of Idators
+     * allows access to stashed values within any defined checks
+     */
+    final Map stashed  //kind of like the Borg pattern
+
+    final Map<String, Closure> extraChecks = [:]
+
+    /**
+     * The default key for any ResultMaps from any Checks defined for this Idators
+     */
+    String resultKey = 'error'
+
+    //The check created by the Definition, target delegate for evaluation/call
+    private Check mCheck
+
+    //List of checks which are defined within the Definition
+    private List definedChecks = new LinkedList()
+    private List requiredChecks = new LinkedList()
+
+    /**
+     * Construct a new Idator using provided arguments and definition
+     * @param checkers Checkers to be used as delegate for Check construction
+     * @param stashed Stash of values to be used across related Idators
+     * @param key Default key for ResultMap results from defined Checks
+     */
+    public Idator(Checkers checkers = new val.Checkers(),
+                  Map stashed = [:],
+                  String key = 'root'){
+        this.checkers = checkers
+        this.stashed = stashed
+        this.resultKey = key
+    }
```

Comment:
set defaults so that the nullary constructor provides the same creation as the previous `Idator` class/factory